src/mapgen.h

```diff
@@ -169,6 +169,17 @@ class jmapgen_piece
         virtual bool is_nop() const {
             return false;
         }
+        /** The pieces will be applied in order of phases.  The phases are as
+         * follows:
+         * -2 - terrain
+         * -1 - furniture
+         *  0 - everything else
+         *  1 - nested mapgen
+         *  2 - transforms and faction ownership
+         */
+        virtual int phase() const {
+            return 0;
```

Comment:
Maybe we should use some constants instead of magic numbers?

Reply:
Seems like a perfect use case for a little enum.